core/src/com/unciv/ui/worldscreen/WorldScreen.kt

```diff
@@ -777,43 +725,45 @@ class WorldScreen(
                 NextTurnAction("Next unit", Color.LIGHT_GRAY,
                     "NotificationIcons/NextUnit") { switchToNextUnit() }
 
-            !game.settings.automatedUnitsMoveOnTurnStart && !viewingCiv.hasMovedAutomatedUnits && viewingCiv.getCivUnits()
-                .any { it.currentMovement > Constants.minimumMovementEpsilon && (it.isMoving() || it.isAutomated() || it.isExploring()) } ->
+            !game.settings.automatedUnitsMoveOnTurnStart && viewingCiv.canMoveAutomatedUnits() ->
                 NextTurnAction("Move automated units", Color.LIGHT_GRAY,
-                    "NotificationIcons/MoveAutomatedUnits") {
-                    viewingCiv.hasMovedAutomatedUnits = true
-                    isPlayersTurn = false // Disable state changes
-                    nextTurnButton.disable()
-                    Concurrency.run("Move automated units") {
-                        for (unit in viewingCiv.getCivUnits())
-                            unit.doAction()
-                        launchOnGLThread {
-                            shouldUpdate = true
-                            isPlayersTurn = true //Re-enable state changes
-                            nextTurnButton.enable()
-                        }
-                    }
-                }
+                    "NotificationIcons/MoveAutomatedUnits") { doAutomatedMoves() }
 
             else ->
                 NextTurnAction("Next turn", Color.WHITE,
-                    "NotificationIcons/NextTurn") {
-                    val action = {
-                        game.settings.addCompletedTutorialTask("Pass a turn")
-                        nextTurn()
-                    }
-                    if (game.settings.confirmNextTurn) {
-                        ConfirmPopup(this, "Confirm next turn", "Next turn", true, action = action).open()
-                    } else {
-                        action()
-                    }
-                }
+                    "NotificationIcons/NextTurn") { doNextTurn() }
+        }
+    }
+
+    private fun doAutomatedMoves() {
+        isTurnActive = false
+        nextTurnButton.disable()
+        Concurrency.run("Move automated units") {
+            viewingCiv.doMoveAutomatedUnits()
+            launchOnGLThread {
+                shouldUpdate = true
+                isTurnActive = true
```

Comment:
What we *really* need here is to change canChangeState to a field, auto-set to 'false' if not player turn or if spectator.
Then, we can change that value manually, without it impacting other display issues.